src/Microsoft.ML.Featurizers/DateTimeTransformer.cs

```diff
@@ -0,0 +1,834 @@
+            public HolidayList Country = HolidayList.None;
+        }
+
+        #endregion
+
+        internal DateTimeEstimator(IHostEnvironment env, string inputColumnName, string columnPrefix, ColumnsProduced[] columnsToDrop, HolidayList country = HolidayList.None)
+        {
+
+            Contracts.CheckValue(env, nameof(env));
+            _host = Contracts.CheckRef(env, nameof(env)).Register("DateTimeTransformerEstimator");
+            _host.CheckValue(inputColumnName, nameof(inputColumnName), "Input column should not be null.");
+
+            _options = new Options
+            {
+                Source = inputColumnName,
+                Prefix = columnPrefix,
+                ColumnsToDrop = columnsToDrop == null ? Array.Empty<ColumnsProduced>() : columnsToDrop,
+                Country = country
+            };
+        }
+
+        internal DateTimeEstimator(IHostEnvironment env, Options options)
+        {
+
+            Contracts.CheckValue(env, nameof(env));
+            _host = Contracts.CheckRef(env, nameof(env)).Register("DateTimeTransformerEstimator");
+
+            _options = options;
+            _options.ColumnsToDrop = _options.ColumnsToDrop == null ? Array.Empty<ColumnsProduced>() : _options.ColumnsToDrop;
+        }
+
+        public DateTimeTransformer Fit(IDataView input)
+        {
+            return new DateTimeTransformer(_host, _options.Source, _options.Prefix, _options.ColumnsToDrop, _options.Country);
+        }
+
+        public SchemaShape GetOutputSchema(SchemaShape inputSchema)
+        {
+            var columns = inputSchema.ToDictionary(x => x.Name);
+
+            foreach (ColumnsProduced column in Enum.GetValues(typeof(ColumnsProduced)))
+                if (_options.ColumnsToDrop == null || !_options.ColumnsToDrop.Contains(column))
+                {
+                    columns[_options.Prefix + column.ToString()] = new SchemaShape.Column(_options.Prefix + column.ToString(), SchemaShape.Column.VectorKind.Scalar,
+                    ColumnTypeExtensions.PrimitiveTypeFromType(column.GetRawColumnType()), false, null);
+                }
+
+            return new SchemaShape(columns.Values);
+        }
+
+        #region Enums
+        public enum ColumnsProduced : byte
+        {
+            Year = 1,
+            Month = 2,
+            Day = 3,
+            Hour = 4,
+            Minute = 5,
+            Second = 6,
+            AmPm = 7,
+            Hour12 = 8,
+            DayOfWeek = 9,
+            DayOfQuarter = 10,
+            DayOfYear = 11,
+            WeekOfMonth = 12,
+            QuarterOfYear = 13,
+            HalfOfYear = 14,
+            WeekIso = 15,
+            YearIso = 16,
+            MonthLabel = 17,
+            AmPmLabel = 18,
+            DayOfWeekLabel = 19,
+            HolidayName = 20,
+            IsPaidTimeOff = 21
+        };
+
+        public enum HolidayList : uint
+        {
+            None = 1,
+            Argentina = 2,
```

Comment:
What about choosing the ISO 3166 country codes for the enum values? 
[https://en.wikipedia.org/wiki/List_of_ISO_3166_country_codes](https://en.wikipedia.org/wiki/List_of_ISO_3166_country_codes)

Reply:
The name of the country needs to be passed to the native code, so I use C# to turn the name of the enum into a string, and then pass it in that way.